src/main/java/com/desiato/puresynth/exceptions/ValidationException.java

```diff
@@ -0,0 +1,13 @@
+package com.desiato.puresynth.exceptions;
+
+import java.util.List;
+
+public class ValidationException extends RuntimeException{
+
+    private final List<String> errorMessages;
```

Comment:
Maybe it would be nice to have a `record` encapsulating an error message, and not operate just on a string.